track_analyzing/utils.hpp

```diff
@@ -0,0 +1,86 @@
+#pragma once
+
+#include "track_analyzing/exceptions.hpp"
+#include "track_analyzing/track.hpp"
+
+#include "routing/geometry.hpp"
+#include "routing/num_mwm_id.hpp"
+
+#include <boost/filesystem.hpp>
+
+#include <cstdint>
+#include <memory>
+#include <string>
+#include <vector>
+
+namespace tracking
+{
+double CalcSubtrackLength(MatchedTrack const & track, size_t begin, size_t end,
+                          routing::Geometry & geometry);
+double CalcTrackLength(MatchedTrack const & track, routing::Geometry & geometry);
+double CalcSpeedKMpH(double meters, uint64_t secondsElapsed);
+bool IsFiltered(std::string const & argument, std::string const & variable);
+void ReadTracks(std::shared_ptr<routing::NumMwmIds> numMwmIds, std::string const & filename,
+                MwmToMatchedTracks & mwmToMatchedTracks);
+MatchedTrack const & GetMatchedTrack(MwmToMatchedTracks const & mwmToMatchedTracks,
+                                     routing::NumMwmIds const & numMwmIds,
+                                     std::string const & mwmName, std::string const & user,
+                                     size_t trackIdx);
+
+template <typename MwmToTracks, typename ToDo>
+void ForTracksSortedByMwmName(ToDo && toDo, MwmToTracks const & mwmToTracks,
+                              routing::NumMwmIds const & numMwmIds)
+{
+  std::vector<std::string> mwmNames;
+  mwmNames.reserve(mwmToTracks.size());
+  for (auto const & it : mwmToTracks)
+    mwmNames.push_back(numMwmIds.GetFile(it.first).GetName());
+  sort(mwmNames.begin(), mwmNames.end());
+
+  for (auto const & mwmName : mwmNames)
+  {
+    auto const mwmId = numMwmIds.GetId(platform::CountryFile(mwmName));
+    auto mwmIt = mwmToTracks.find(mwmId);
+    CHECK(mwmIt != mwmToTracks.cend(), ());
+    toDo(mwmName, mwmIt->second);
+  }
+}
+
+template <typename ToDo>
+void ForEachTrackFile(ToDo && toDo, std::string const & filepath, std::string const & extension,
+                      shared_ptr<routing::NumMwmIds> numMwmIds)
+{
+  if (!boost::filesystem::exists(filepath.c_str()))
```

Comment:
For example, on our dev-servers it's a quiz to find not only boost, but even modest g++.